lib/users.js

```diff
@@ -14,23 +17,162 @@ const db = require('./db')
  * @returns {user object} User object (containing name)
  */
 function get (token) {
-  const sql = 'select * from user_tokens where token=$1'
+  const sql = 'SELECT * FROM user_tokens WHERE token=$1'
   return db.oneOrNone(sql, [token])
 }
 
 /**
- * Get multiple users given an array of tokens
+ * Get multiple users given an array of ids
  *
  * @name getByIds
  * @function
  *
- * @param {array} tokens Array with users' tokens
+ * @param {array} ids Array with users' ids
  *
  * @returns {array} Array of users found
  */
-function getByIds (tokens) {
-  const sql = 'select * from user_tokens where token in ($1^)'
-  const tokensClause = _.map(pgp.as.text, tokens).join(',')
-  return db.any(sql, [tokensClause])
+function getByIds (ids) {
+  const sql = `SELECT * FROM users WHERE id IN ($1^)`
+  const idList = _.map(pgp.as.text, ids).join(',')
+  return db.any(sql, [idList])
+}
+
+function getUsers () {
+  const sql = `SELECT id, username, role, enabled, last_accessed, last_accessed_from, last_accessed_address FROM users ORDER BY username`
+  return db.any(sql)
+}
+
+function getByName (username) {
+  const sql = `SELECT id, username, role, last_accessed FROM users WHERE username=$1 limit 1`
+  return db.oneOrNone(sql, [username])
+}
+
+function verifyAndUpdateUser (id, ua, ip) {
+  const sql = `SELECT id, username, role, enabled FROM users WHERE id=$1 limit 1`
+  return db.oneOrNone(sql, [id]).then(user => {
+    if (!user) return null
+
+    const sql2 = `UPDATE users SET last_accessed=now(), last_accessed_from=$1, last_accessed_address=$2 WHERE id=$3 RETURNING id, role, enabled`
+    return db.one(sql2, [ua, ip, id]).then(user => {
+      return user
+    })
+  })
+}
+
+function createUser (username, password, role) {
+  const sql = `INSERT INTO users (id, username, password, role) VALUES ($1, $2, $3, $4)`
+  bcrypt.hash(password, 12).then(function (hash) {
+    return db.none(sql, [uuid.v4(), username, hash, role])
+  })
+}
+
+function deleteUser (id) {
+  const sql = `DELETE FROM users WHERE id=$1`
```

Comment:
We don't want to hard delete the users. Probably best to disable them and hide from ui and services

Reply:
This function is not being used, might as well delete it